### DISCUSSION TITLE
[type-safe-api] Add support for Integrations.AWS()

### DISCUSSION THREAD

**Discussion by v41e** [aws/aws-pdk#539] [Ideas]
It would be nice to have `Integration.AWS({options})` support, that way it might be possible to route an API endpoint to an S3 location for example.

**Comment by cogwirrel** [comment]
It'd definitely be a handy feature to have more built in integrations than just `Integrations.lambda`! My feeling is they should be more specific though if possible, so rather than `Integrations.AWS` we have `Integrations.s3`, `Integrations.sqs`, etc.

Feel free to send a PR for `Integrations.s3` if you like! :)

Note also [my answer here](https://github.com/aws/aws-prototyping-sdk/discussions/541#discussioncomment-6770936) re custom integrations, which allow you to implement any API gateway supported integration in your own codebase too :)

**Reply by v41e** [reply]
i'll definitely open a PR for this integration :)

**Reply by cogwirrel** [reply]
That would be brilliant! 🙌
